src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs

```diff
@@ -147,7 +147,7 @@ public int Execute(params string[] args)
                             if (!option.TryParse(longOption[1]))
                             {
                                 command.ShowHint();
-                                throw new Exception(string.Format("TODO: Error: unexpected value '{0}' for option '{1}'", longOption[1], option.LongName));
+                                throw new CommandParsingException(command, $"Error: unexpected value '{longOption[1]}' for option '{option.LongName}'");
```

Comment:
Remove the "Error:" from the front and capitalize "unexpected". "Error" is a prefix that catcher of the exception should control.